python/cudf/bindings/replace.pyx

```diff
@@ -20,15 +22,28 @@ from libc.stdint cimport uintptr_t
 from libc.stdlib cimport calloc, malloc, free
 
 
-cpdef replace(col, old_values, new_values):
+cpdef replace(input_col, values_to_replace, replacement_values):
     """
-        Call gdf_find_and_replace_all
+        Call cudf::find_and_replace_all
     """
-    cdef gdf_column* c_col = column_view_from_column(col)
-    cdef gdf_column* c_old_values = column_view_from_column(old_values)
-    cdef gdf_column* c_new_values = column_view_from_column(new_values)
+    cdef gdf_column* c_input_col = \
+                            column_view_from_column(input_col)
+    cdef gdf_column* c_values_to_replace = \
+                            column_view_from_column(values_to_replace)
+    cdef gdf_column* c_replacement_values = \
```

Comment:
These are leaking, you need to free the `gdf_column` objects created here before returning at the end.

Reply:
Sure, I'll fix it.

Reply:
I made a commit with changes to solve memory leaking https://github.com/rapidsai/cudf/pull/1696/commits/7eba55cc3994b839dafa43c3d68783f31af65246 but then other tests were failing i.e. `test_groupby_as_index_multiindex()` and others

So, i commented those lines in https://github.com/rapidsai/cudf/pull/1696/commits/c26013e8fe4639cea3c2c1a7c98fef423a27a876 then everything seems to be working

Reply:
> I made a commit with changes to solve memory leaking [7eba55c](https://github.com/rapidsai/cudf/commit/7eba55cc3994b839dafa43c3d68783f31af65246) but then other tests were failing i.e. `test_groupby_as_index_multiindex()` and others
> 
> So, i commented those lines in [c26013e](https://github.com/rapidsai/cudf/commit/c26013e8fe4639cea3c2c1a7c98fef423a27a876) then everything seems to be working

If we comment out those frees then we're still leaking memory. Sounds like there's an issue somewhere in the groupby code that needs to be addressed.

Reply:
Nevermind, instead of using `gdf_column_free` you want to actually just free the struct using `free` directly. We don't want libcudf to free the underlying memory / mask allocations.

Reply:
sure, will do it.

Reply:
Can you please have a look at it now? I have committed the latest changes as per your suggestion.